src/data/option/basic.lean

```diff
@@ -71,6 +71,9 @@ by cases x; [contradiction, rw get_or_else_some]
 theorem mem_unique {o : option α} {a b : α} (ha : a ∈ o) (hb : b ∈ o) : a = b :=
 option.some.inj $ ha.symm.trans hb
 
+theorem eq_of_mem_of_mem {a : α} {o1 o2 : option α} (h1 : a ∈ o1) (h2 : a ∈ o2) : o1 = o2 :=
+by { rw option.mem_def at *, exact eq.trans h1 h2.symm }
```

Comment:
```suggestion
h1.trans h2.symm
```
also works.

Reply:
Oh, I only just saw the Zulip discussion... sorry for the noise!

Reply:
No problem, thanks!